src/Agent.Worker/TaskDefinitionManager.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Handlers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    [ServiceLocator(Default = typeof(TaskDefinitionManager))]
+    public interface ITaskDefinitionManager : IAgentService
+    {
+        Task DownloadAsync(IExecutionContext executionContext, IEnumerable<TaskInstance> tasks);
+
+        Definition Load(TaskReference task);
+    }
+
+    public sealed class TaskDefinitionManager : AgentService, ITaskDefinitionManager
+    {
+        private async Task DownloadAsync(IExecutionContext executionContext, TaskReference task)
```

Comment:
These methods are mostly the same. I'm pretty sure I made some small changes (e.g. tracing, rename some variables, small cleanups), but this is largely intact.

I removed getting the task directory as a public method. It turned out to not be needed as a separate method. I made the new Load method return the path and the definition data in one wrapper object.